src/model/mod.rs

```diff
@@ -530,10 +530,16 @@ impl<'a, A: AlleleFreqs, B: AlleleFreqs, P: priors::PairModel<A, B>> PairPileup<
     ) -> LogProb {
         // no af_control given, because case and control are independent
         if af_control.is_none() {
+            let l: LogProb;
+            if let Some(likelihood) = self.case_likelihood_cache.get(&af_case) {
                 // get likelihood if already cached
-            *self.case_likelihood_cache.entry(af_case)
-                // compute and insert otherwise
-                .or_insert( self.case_sample_model.likelihood_pileup(&self.case, af_case, None) )
+                return *likelihood
+            } else {
+                // compute otherwise
+                l = self.case_sample_model.likelihood_pileup(&self.case, af_case, None);
+            }
+            self.case_likelihood_cache.insert(af_case, l);
+            l
```

Comment:
Does this not work?
```
if let Some(likelihood) = self.case_likelihood_cache.get(&af_case)  {
    return likelihood;
}
let l = self.case_sample_model.likelihood_pileup(&self.case, af_case, None);
self.case_likelihood_cache.insert(af_case, l);
l
```

Reply:
It does, and it is shorter. So tests are running!